src/Utils/Github.hs

```diff
@@ -0,0 +1,78 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Utils.Github
+  ( showGithubException
+  , githubAPILimitErrorText
+  )
+where
+
+import Data.Aeson (FromJSON, ToJSON, decode)
+import Data.ByteString.Char8 (ByteString)
+import Data.ByteString.Lazy (fromStrict)
+import Data.Text (Text, isPrefixOf)
+import GHC.Generics (Generic)
+import qualified Network.HTTP.Client as Client
+import qualified Network.HTTP.Req as Req
+import PyF (fmt)
+
+import Utils.Req (showHTTPException, showRawResponse)
+
+-- | Represents a typical Github Error serialized as JSON like so:
+--
+-- @
+-- {
+--    "message": "the error reason"
+-- }
+-- @
+data GithubError = GithubError
+  { message :: Text
+  } deriving (Generic, Show)
+
+instance ToJSON GithubError where
+    -- No need to provide a toJSON implementation.
+
+instance FromJSON GithubError
+    -- No need to provide a parseJSON implementation.
+
+-- | Uses the helper to show generic HTTP issues and provides a specific handler for Github
+-- "business" exceptions
+showGithubException ::
+  Req.HttpException ->
+  Text
+showGithubException = showHTTPException handleGithubException
+
+handleGithubException ::
+  Client.Response () ->
+  ByteString ->
+  Text
+handleGithubException resp body =
+  case err of
+    Just githubError -> showRawGithubMessage githubError
+    Nothing -> showRawResponse resp body
+  where err = decode $ fromStrict body ::Maybe GithubError
```

Comment:
For information. For this kind of stuff, you can use `maybe` from `Data.maybe` which may or may not more readable. Something like:

```
maybe (showRawResponse resp body) showRawGithubMessage err
```

Reply:
blah....
Undecided. I'll change if you feel strongly about it

Reply:
No, I don't care. Just that we had an offline discussion about this, so I just wanted to make you aware of this way of doing things. I personally prefere the `case` approach which I find more general.